optaplanner-spring-integration/optaplanner-spring-boot-autoconfigure/src/main/java/org/optaplanner/spring/boot/autoconfigure/OptaPlannerAutoConfiguration.java

```diff
@@ -200,33 +201,69 @@ private void applySolverProperties(SolverConfig solverConfig) {
         return new ArrayList<>(entityClassSet);
     }
 
-    private void applyScoreDirectorFactoryProperties(SolverConfig solverConfig) {
+    protected void applyScoreDirectorFactoryProperties(SolverConfig solverConfig) {
+        Optional<String> constraintsDrlFromProperty = constraintsDrl();
+        Optional<String> defaultConstraintsDrl = defaultConstraintsDrl();
+        Optional<String> effectiveConstraintsDrl = constraintsDrlFromProperty.map(Optional::of).orElse(defaultConstraintsDrl);
```

Comment:
(non blocking)
Personally, I am not a fan of using Optional outside of Streams - a nullable var works just as well - see Stuart's and Goetz's talks around this topic.